process/interceptors/multiDataInterceptor.go

```diff
@@ -199,7 +206,29 @@ func (mdi *MultiDataInterceptor) ProcessReceivedMessage(message p2p.MessageP2P,
 		mdi.throttler.EndProcessing()
 	}()
 
-	return nil
+	messageID := mdi.createInterceptedMultiDataMsgID(listInterceptedData)
+
+	return messageID, nil
+}
+
+func (mdi *MultiDataInterceptor) createInterceptedMultiDataMsgID(interceptedMultiData []process.InterceptedData) []byte {
+	if len(interceptedMultiData) == 0 {
+		return nil
+	}
+	if len(interceptedMultiData) == 1 {
+		return interceptedMultiData[0].Hash()
+	}
+
+	lenOneID := len(interceptedMultiData[0].Hash())
+	data := make([]byte, 0, lenOneID*len(interceptedMultiData))
+	for _, id := range interceptedMultiData {
+		data = append(data, id.Hash()...)
+	}
+	if len(data) == 0 {
+		return nil
```

Comment:
this has a side effect that you return nil, nil un the above function.

Reply:
changed